libraries/bezier-rs/src/subpath/manipulators.rs

```diff
@@ -0,0 +1,114 @@
+use super::*;
+use crate::ComputeType;
+
+impl Subpath {
+	/// Calculate the point on the subpath based on the parametric `t`-value provided.
+	/// Expects `t` to be within the inclusive range `[0, 1]`.
+	pub fn add_manipulator_group(&mut self, t: ComputeType) {
+		match t {
+			ComputeType::Parametric(t) => {
+				assert!((0.0..=1.).contains(&t));
+
+				let number_of_curves = self.len_segments() as f64;
+				let scaled_t = t * number_of_curves;
+
+				let target_curve_index = scaled_t.floor() as i32;
+				let target_curve_t = scaled_t % 1.;
+
+				if target_curve_t == 0. || target_curve_t == 1. {
```

Comment:
Do you think we should check these conditions to some epsilon rather than strict equality?